pkg/image-manager/controller.go

```diff
@@ -94,6 +96,42 @@ func (h *baseImageEntityHandler) Error(ctx context.Context, obj entitystore.Enti
 type imageEntityHandler struct {
 	Store   entitystore.EntityStore
 	Builder *ImageBuilder
+	// TODO (bjung): This is a temporary workaround for concurrency issues
+	// while building images.  A holistic solution for all controllers is
+	// needed.
+	lock    sync.Map
+	timeout time.Duration
+}
+
+// CheckAndLock checks to see if the entity is currently being worked on if not
+// stores the entity
+func (h *imageEntityHandler) CheckAndLock(obj entitystore.Entity) bool {
+	// Only do a single image create at a time (for a given image)
+	now := time.Now()
+	v, loaded := h.lock.LoadOrStore(obj.GetID(), now)
+	if loaded {
```

Comment:
NIT: I know it's a matter of preference, but handling `if !loaded` first would reduce the overall indentation. Same with nested if, returning false in `if duration > h.timeout` would remove the `else` clause and increase readability.